File: cpp/cdmsbats/chargechanrecord.cpp
#include <iostream>
#include <iomanip>

#include "chargechanrecord.h"
#include "datarecord.h"

using namespace std;

ChargeChanRecord::ChargeChanRecord()
{
    _detCode = NULL;
    _towerNum = NULL;
    _driverGain = NULL;
    _chanBias = NULL;
    _rtfOffset = NULL;
    _deltat = NULL;
    _t0 = NULL;
    _traceLength = NULL;
}

void ChargeChanRecord::StoreValues(int32_t* Record)
{
    _detCode = Record[0];
    _towerNum = Record[1];
    _driverGain = float(Record[2])/100;
    _chanBias = float(Record[3])/1e6;
    _rtfOffset = float(Record[4])/1e6;
    _deltat = float(Record[5])/1000;
    _t0 = float(Record[6])/1000;
    _traceLength = Record[7];
}

void ChargeChanRecord::PrintValues()
{
    cout << "Detector Code = " << _detCode << endl;
    cout << "Tower Number = " << _towerNum << endl;
    cout << "Driver Gain = " << fixed << setprecision(1) << _driverGain << endl;
    cout << "Channel Bias = " << fixed << setw(5) << setprecision(3) << _chanBias << " V" << endl;
    cout << "RTF Offset = " << fixed << setw(5) << setprecision(3) << _rtfOffset << " V" << endl;
    cout << "Delta T = " << fixed << setw(4) << setprecision(2) << _deltat << " us" << endl;
    cout << "t0 = " << fixed << setw(5) << setprecision(1) << _t0 << " us" << endl;
    cout << "Trace Length = " << _traceLength << endl;
}

File: compiled/julia/cdmsbats/ChargeChanRecord.jl
include("/home/tdoughty1/Workspace/programming/julia/cdmsbats/DataRecord.jl")

type ChargeChanRecord <: DataRecord

    _detCode
    _towerNum
    _driverGain
    _chanBias
    _rtfOffset
    _deltat
    _t0
    _traceLength

    ChargeChanRecord() = new(None, None, None, None, None, None, None, None)
end

function StoreValues(recPtr::ChargeChanRecord, Record::Array)

    recPtr._detCode = Record[1] & 0xFFFFFFFF
    recPtr._towerNum = Record[2]
    recPtr._driverGain = float(Record[3])/100
    recPtr._qetBias = float(Record[4])/100
    recPtr._rtfOffset = float(Record[5])/1e6
    recPtr._deltat = float(Record[6])/1000
    recPtr._t0 = float(Record[7])/1000
    recPtr._traceLength = Record[8]

end

function PrintValues(recPtr::ChargeChanRecord)

    @printf("Detector Code = %d\n", recPtr._detCode)
    @printf("Tower Number = %d\n", recPtr._towerNum)
    @printf("Driver Gain = %.1f\n", recPtr._driverGain)
    @printf("Channel Bias = %6.2f\n", recPtr._chanBias)
    @printf("RTF Offset = %4.2f pA\n", recPtr._rtfOffset)
    @printf("Delta T = %4.2f us V\n", recPtr._deltat)
    @printf("t0 = %5.1f us\n", recPtr._t0)
    @printf("Trace Length = %d\n", recPtr._traceLength)

end
